OutOfSchool/OutOfSchool.WebApi/Startup.cs

```diff
@@ -392,5 +395,11 @@ public static void AddApplicationServices(this WebApplicationBuilder builder)
             .AddDbContextCheck<OutOfSchoolDbContext>(
                 "Database",
                 tags: new[] { "readiness" });
+
+        services.AddHeaderPropagation(options =>
+        {
+            options.Headers.Add("Request-Id");
```

Comment:
(Logic) Don't you want to add some default value builder? (As a result for case when `Request-Id` is empty on one step you could rehidrate it by some GUID and track all subsequent steps)